test/hash/src/test_hash.d

```diff
@@ -87,6 +88,35 @@ void issue19005() @nogc nothrow pure @safe
     auto hash = date.hashOf;
 }
 
+/// Accept SIMD vectors.
+void issue19204() @nogc nothrow pure @safe
+{
+    version (D_SIMD)
+    {
+        static import simd = core.simd;
+        static if (is(simd.int4)) // __traits(isArithmetic)
+        {{
+            enum simd.int4 val = [1,2,3,4];
+            enum ctfeHash = hashOf!(simd.int4)(val);
+            simd.int4 rtVal = val;
+            auto rtHash = hashOf(rtVal);
+            assert(ctfeHash == rtHash);
+        }}
+        static if (is(simd.void16)) // non __traits(isArithmetic)
+        {{
+            auto h = hashOf(simd.void16.init);
+        }}
+        static if (is(simd.float4)) // __traits(isArithmetic) and __traits(isFloating)
+        {{
+            enum simd.float4 val = [1.1f, 2.2f, 3.3f, 4.4f];
+            enum ctfeHash = hashOf!(simd.float4)(val);
```

Comment:
Why do you need to explicitly instantiate `hashOf` here?

Reply:
Didn't need to.

Reply:
Actually, I did need to. It was long enough ago that I forgot. Without the explicit instantiation the type is inferred as a floating point array.